File: 프로그래머스/2/154538. 숫자 변환하기/숫자 변환하기.js
function solution(x, y, n) {
    var answer = -1;
    var que = [];
    que.push([y, 0])
    
    while(que.length){
        var now = que.shift();
        if(now[0] < x) continue;
        if(now[0] == x){
            answer = now[1];
            break;
        }
        if(now[0]%3 == 0){
            que.push([now[0]/3, now[1]+1])
        }
        if(now[0]%2 == 0){
            que.push([now[0]/2, now[1]+1])
        }
        que.push([now[0]-n, now[1]+1])
    }
    
    return answer;
}

File: 프로그래머스/2/154538. 숫자 변환하기/숫자 변환하기.java
import java.util.*;

class Solution {
     
    public int solution(int x, int y, int n) {
        
        int answer = -1;
        Deque<int[]> que = new ArrayDeque<>();
        que.add(new int[]{y, 0});
        
        while(!que.isEmpty()){
            int now[] = que.poll();
            int nowY = now[0];
            if(nowY < x) continue;
            if(nowY == x){
                answer = now[1];
                break;
            }
            if(nowY%3 == 0){
                que.add(new int[]{nowY/3, now[1]+1});
            }
            if(nowY%2 == 0){
                que.add(new int[]{nowY/2, now[1]+1});
            }
            que.add(new int[]{nowY-n, now[1]+1});
        }
       
        return answer;
    }
    
}